### PR TITLE
Shared object creation for amd64 and aarch64

### PR DESCRIPTION
**Pull request by kczulko**
# Goal
The goal of this PR is the release for aarch64 platform.

# Discussion
- adds aarch64 as release target for jni *.so file
- possible missing of other architectures: x86_64-apple-darwin, x86_64-pc-windows-gnu, aarch64-apple-darwin

# Checklist
- [ ] Tests added

### CODE REVIEW

**Comment by kczulko** [comment]
Oh. I see now that there's a lot to do within .github/workflows/release.yml

I've used nix to provide aarch64 linker and I think this might be an interesting alternative compared to os matrix:
```
{ pkgs ? import <nixpkgs> {}}:
with pkgs; mkShell {
  CARGO_TARGET_AARCH64_UNKNOWN_LINUX_GNU_LINKER =
    let
      inherit (pkgsCross.aarch64-multiplatform.stdenv) cc;
    in
      "${cc}/bin/${cc.targetPrefix}cc";
  
  buildInputs = [ rustup ];
}
```

This "shop" contains also:
- `pkgs.pkgsCross.mingwW64` (which may fit windows build)
- `pkgs.pkgsCross.aarch64-darwin`
- `pkgs.pkgsCross.x86_64-darwin`

Together with [magic nix cache](https://determinate.systems/posts/magic-nix-cache/) and [nix installer](https://github.com/DeterminateSystems/nix-installer-action) build can be quite fast.

File: java/lib/src/main/java/io/amplica/graphsdk/Native.java [hunk 1]
```diff
@@ -11,10 +11,24 @@ public final class Native {
...
 
+    private static String obtainPlatform() {
+        String os = System.getProperty("os.name", "unknown");
+        String arch = System.getProperty("os.arch", "unknown");
+
+        if (os.contains("nux") && arch.contains("amd64")) {
+            return "x86_64-unknown-linux-gnu";
+        } else if (os.contains("nux") && arch.contains("aarch64")) {
+            return "aarch64-unknown-linux-gnu";
+        } else {
```
Comment (enddynayn) [inline java/lib/src/main/java/io/amplica/graphsdk/Native.java:22]: I think we can remove the else here.

**Review by aramikm — COMMENTED** [review]
@kczulko Thanks for opening this PR but as you mentioned this would break all other platforms. I think to unblock you guys immediately and temporarily you can fork this repo if you want and publish your changes so that might fix the issue on your platform but to fix it for all platforms it requires more changes that we will try to take care of it

**Comment by kczulko** [comment]
Thx @aramikm. I've checked one thing: using nix, together with this shell.nix file, I am able to build almost everything on my platform (x86_64-linux, Nixos). The only missing thing is any darwin related, as this is not yet supported in terms of nix cross compilation on x86_64-linux host platform:
```
{ pkgs ? import <nixpkgs> {}}:

let
  mkLinkerPath = stdenv:
    let
      inherit (stdenv) cc;
    in
      "${cc}/bin/${cc.targetPrefix}cc";

  pthreadPkg = { stdenv, lib, fetchzip}: stdenv.mkDerivation {
    pname = "pthreads-w32";
    version = "2.9.1";

    src = fetchzip {
      url = "https://sourceware.org/pub/pthreads-win32/pthreads-w32-2-9-1-release.tar.gz";
      sha256 = "1s8iny7g06z289ahdj0kzaxj0cd3wvjbd8j3bh9xlg7g444lhy9w";
    };

    makeFlags = [ "CROSS=${stdenv.cc.targetPrefix}" "GC-static" ];

    installPhase = ''
      runHook preInstall

      install -D libpthreadGC2.a $out/lib/libpthread.a

      runHook postInstall
    '';
  };
  
  pthread = pkgs.pkgsCross.mingwW64.callPackage pthreadPkg {};
in

with pkgs; mkShell {
  CARGO_TARGET_AARCH64_UNKNOWN_LINUX_GNU_LINKER = mkLinkerPath pkgsCross.aarch64-multiplatform.stdenv;
  CARGO_TARGET_X86_64_UNKNOWN_LINUX_GNU_LINKER = mkLinkerPath pkgsCross.gnu64.stdenv;
  # CARGO_TARGET_AARCH64_APPLE_DARWIN_LINKER = mkLinkerPath pkgsCross.aarch64-darwin.stdenv;
  CARGO_TARGET_X86_64_PC_WINDOWS_GNU_LINKER = mkLinkerPath pkgsCross.mingwW64.stdenv;


  PTHREAD_LOCATION="${pthread}/lib";
  buildInputs = [ rustup ];
}
```

And here's what I've got in the java/lib/src/main/resources:
```
[nix-shell:~/Projects/graph-sdk/java/lib/src/main/resources]$ tree
.
├── aarch64-unknown-linux-gnu
│   └── libdsnp_graph_sdk_jni.so
├── x86_64-pc-windows-gnu
│   └── dsnp_graph_sdk_jni.dll
└── x86_64-unknown-linux-gnu
    └── libdsnp_graph_sdk_jni.so

4 directories, 3 files
```

**Review by aramikm — COMMENTED** [review]
Looks really interesting, I wish it had support for apple darwin then we could simplify everything but for now I guess we have to stick with github-runners. Another thing that I'm also looking into is [cross](https://crates.io/crates/cross) library which looks promising but until we test it we are not sure it would work with all desired platforms.

**Comment by aramikm** [comment]
@kczulko We released graph-sdk `v1.0.2` which has multi-target support. (Including linux-aarch64). I'm going to close this PR and thanks for your help.

**Pull request closed without merging**